pkg/services/sqlstore/migrations/ualert/channel.go

```diff
@@ -8,23 +8,21 @@ import (
 	"sort"
 	"strings"
 
-	"github.com/prometheus/alertmanager/pkg/labels"
-
-	"github.com/grafana/grafana/pkg/components/securejsondata"
 	"github.com/grafana/grafana/pkg/components/simplejson"
 	"github.com/grafana/grafana/pkg/util"
+	"github.com/prometheus/alertmanager/pkg/labels"
 )
 
 type notificationChannel struct {
-	ID                    int64                         `xorm:"id"`
-	OrgID                 int64                         `xorm:"org_id"`
-	Uid                   string                        `xorm:"uid"`
-	Name                  string                        `xorm:"name"`
-	Type                  string                        `xorm:"type"`
-	DisableResolveMessage bool                          `xorm:"disable_resolve_message"`
-	IsDefault             bool                          `xorm:"is_default"`
-	Settings              *simplejson.Json              `xorm:"settings"`
-	SecureSettings        securejsondata.SecureJsonData `xorm:"secure_settings"`
+	ID                    int64            `xorm:"id"`
+	OrgID                 int64            `xorm:"org_id"`
+	Uid                   string           `xorm:"uid"`
+	Name                  string           `xorm:"name"`
+	Type                  string           `xorm:"type"`
+	DisableResolveMessage bool             `xorm:"disable_resolve_message"`
+	IsDefault             bool             `xorm:"is_default"`
+	Settings              *simplejson.Json `xorm:"settings"`
+	SecureSettings        SecureJsonData   `xorm:"secure_settings"`
```

Comment:
Why is this not `map[string][]byte`? (not as in I don't think it should be, but "I don't understand this")

Reply:
Probably because it was like this before refactoring https://github.com/grafana/grafana/blob/main/pkg/services/sqlstore/migrations/ualert/channel.go#L27 🤔

Reply:
Let's leave this one unresolved and @joanlopez can let us know when he's back. This should work as-is.

Reply:
The reason is that we decided to keep the CLI & DB migrations with the previous behavior and code (SecureJsonData) because I guessed the behavior should be the same and envelope encryption don't applies here.

For existing and up-to-date instances, it should be executed already, therefore not changed, and for the other ones, I think these migrations should be executed before the envelope encryption ones (as they're part of the steps to update between older versions), so no envelope encryption changes should affect this.

Does that make sense? 🤔

Reply:
Ah, I think I understand! @undef1nd was going to write a few words about this as well to explain it to me, and I think I'll spend a bit of time trying to really grasp it 🦥😇